### DISCUSSION TITLE
LoadAsync immediately after SaveChangesAsync returns null

### DISCUSSION THREAD

**Discussion by thomaslevesque** [JasperFx/marten#3433] [Q&A] [answered]
Hi there,
Probably a noob question, I have no experience with Marten or with Event Sourcing, just experimenting at the moment.
I have a `TodoProjection` registered as `Inline`, and an API endpoint like this:
```csharp
app.MapPost("todo/new", async (CreateTodoRequest request, IDocumentSession session) =>
{
    var id = Guid.NewGuid();
    var @event = new TodoCreated(id, request.Text, request.DueDate);
    session.Events.StartStream<Todo>(@event);
    await session.SaveChangesAsync();
    return await session.LoadAsync<Todo>(id);
}).WithName("Create Todo");
```
The `Todo` is correctly created (I can get it in a subsequent request), but `LoadAsync` always returns null.

What am I missing?

**Comment by thomaslevesque** [comment]
I also tried to change `session.LoadAsync` to `session.Events.AggregateStreamAsync`, but it also returns null.
How can I get the initial state of the aggregate right after starting the event stream?

**Reply by mysticmind** [reply]
Could you please share your full code to check?

**Reply by thomaslevesque** [reply]
Hi @mysticmind,
Here's my full code: https://gist.github.com/thomaslevesque/cb454b1fdc3cd2661af54ea072ae5f1a
The issue is on line 41 of Program.cs

**Comment by mysticmind** [comment, marked as answer]
@thomaslevesque I figured out the issue, change `session.Events.StartStream<Todo>(@event);` to `session.Events.StartStream<Todo>(id, @event);`. Essentially, we are setting the stream Id same as the Todo Id otherwise the stream will generate it's own Id which will reflect as the Todo Id in the projected document.

**Reply by thomaslevesque** [reply]
Works perfectly now, thanks!
TBH it's a bit surprising. Since Marten can obviously figure out that `Todo.Id` is the stream id (since I can get the projected document later using this id), why doesn't it do it automatically?
Anyway, this fixes the problem. Thanks again!

**Reply by mysticmind** [reply]
Stream `Id` will be the `Todo` `Id` in the doc. If you closely look up the event entries with the problem you had, the `Todo` `Id` which you passed via `var id = Guid.NewGuid();` and the Stream `Id` set would differ in this case. If you look up the `Todo` record Id in db and try to fetch it based on what you see in the db entry, you will get the data. But the `Id` is not the same as what you originally wanted to use (it is an auto generated stream id since you didn't pass one while creating the stream).

**Reply by mysticmind** [reply]
You can double check this, check the `Id` in your event entries, stream and Todo doc. All have to match for the the implementation which you are attempting to do.

Without the fix for `StartStream(...)`:
- Stream and Todo doc Id will match
- Id in event entries (which is your intended Id) won't match with the both Stream/Todo Id's.

**Reply by thomaslevesque** [reply]
I see, thanks for the explanation!
